homeassistant/components/home_connect/switch.py

```diff
@@ -41,7 +41,7 @@ class HomeConnectProgramSwitch(HomeConnectEntity, SwitchEntity):
 
     def __init__(self, device, program_name):
         """Initialize the entity."""
-        desc = " ".join(["Program", program_name.split(".")[-1]])
+        desc = " ".join(["Program", program_name])
         super().__init__(device, desc)
```

Comment:
It looks like desc is used in the base entity to generate the unique id.  If that's the case, will this be a breaking change for existing sensors?

Reply:
I think you're right! I will come up with another solution, because I don't like this being a breaking change.

Reply:
I'm trying to grasp the breaking change here, it is needed to support the WasherDryer class?

Reply:
Because otherwise there will be a clash between washer and dryer programs. See https://github.com/DavidMStraub/homeassistant-homeconnect/issues/119